sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -353,49 +355,49 @@ public void TearDownRecordedTestClass()
             if (Mode == RecordedTestMode.Record)
             {
                 var testClassDirectory = new DirectoryInfo(GetSessionFileDirectory());
-                if (!testClassDirectory.Exists)
+                if (testClassDirectory.Exists)
                 {
-                    return;
-                }
-
-                var knownMethods = new HashSet<string>();
+                    var knownMethods = new HashSet<string>();
 
-                // Management tests record in ctor
-                knownMethods.Add(GetType().Name);
+                    // Management tests record in ctor
+                    knownMethods.Add(GetType().Name);
 
-                // Collect all method names
-                foreach (var method in GetType()
-                             .GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.Instance))
-                {
-                    // TestCase attribute allows specifying a test name
-                    foreach (var attribute in method.GetCustomAttributes(true))
+                    // Collect all method names
+                    foreach (var method in GetType()
```

Comment:
Do we also need to update the documentation for this new scenario? If so, what would be the right place to do it?

/cc: @benbp , @weshaggard do we have central documentation around this for partners? Or each language is making independent calls here?

Reply:
@pallavit Are you referring to `asset-sync` in general? EG externalizing the recordings? if the partners are contributing code to this (or other azure-sdk-for-X) repos, I think they should probably follow what the rest of the folks in this repo are doing and store their data in `azure-sdk-assets`.

Anyone part of the `azure-sdk` umbrella (this includes the azure-sdk-write-X or azure-sdk-push-X teams) has write access to the asets repo.

Other than that:
- java auto downloads the standalone executable and leaves it up to the users to push
- js integrated the migration script + push/restore commands into their utility recorder package (version of azure.core.test)
- python added the proxy push/restore to their local dev script
- C++ run it manually as far as I understand it
- Go does the same